AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -141,6 +142,43 @@ class CompleteScreenViewModel: ObservableObject {
         exitRoute()
     }
     
+    private func getAndDisplayData() {
+        if session.stream.first!.sensorName.contains("AirBeam") {
+            do {
+                try streamsDownloader.downloadStreams(with: session.id, for: .airBeam3) { result in
+                    switch result {
+                    case .success(let success):
+                        // TODO: - FIX Thresholds and Improve this part of code
+                        // TODO: Compare strings and get threshold value, example - comment below
+                        // let vl = MeasurementStream.init(sensorName: .pm1, sensorPackageName: "").thresholdVeryLow
+                        let partialSession = success.map({ PartialExternalSession.Stream(id: $0.streamId,
+                                                                                         unitName: "microgram per cubic meter",
+                                                                                         unitSymbol: $0.sensorUnit,
+                                                                                         measurementShortType: "PM",
+                                                                                         measurementType: "",
+                                                                                         sensorName: $0.sensorName,
+                                                                                         sensorPackageName: "",
+                                                                                         thresholdsValues: .init(veryLow: Int32(0),
+                                                                                                                 low: Int32(30),
+                                                                                                                 medium: Int32(40),
+                                                                                                                 high: Int32(50),
+                                                                                                                 veryHigh: Int32(120)))})
+                        self.session.stream = []
+                        partialSession.forEach { self.session.stream.append($0) }
+                        Log.info("Completed downloading missing streams.")
+                        self.getMeasurementsAndDisplayData()
+                    case .failure(let failure):
+                        Log.info("Something want wrong when downoading missing streams.")
+                    }
+                }
+            } catch {
+                Log.info("Something want wrong when downoading missing streams.")
```

Comment:
Could we have a different message here, so that we would know what went wrong when reading logs?

Reply:
I've added error.localizedDescription so far